src/Raven.Server/ServerWide/CompareExchangeExpirationStorage.cs

```diff
@@ -80,40 +80,18 @@ internal static IEnumerable<(Slice keySlice, long expiredTicks, Slice ticksSlice
             }
         }
 
-        public static bool HasExpiredMetadata(BlittableJsonReaderObject value, out long ticks, Slice keySlice, string storageKey = null)
+        public static bool TryGetExpires(BlittableJsonReaderObject value, out long ticks)
         {
             ticks = default;
-            if (value.TryGetMember(Constants.Documents.Metadata.Key, out var metadata) == false || metadata == null)
+            if (value.TryGetMember(Constants.Documents.Metadata.Key, out var metadata) == false || metadata is not BlittableJsonReaderObject metadataReader)
+                return false;
+        
+            if (metadataReader.TryGet(Constants.Documents.Metadata.Expires, out LazyStringValue expirationDate) == false)
                 return false;
 
-            if (metadata is BlittableJsonReaderObject bjro && bjro.TryGet(Constants.Documents.Metadata.Expires, out object obj))
-            {
-                if (obj is LazyStringValue expirationDate)
-                {
-                    if (DateTime.TryParseExact(expirationDate, DefaultFormat.DateTimeFormatsToRead, CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind,
-                            out DateTime date) == false)
-                    {
-                        storageKey ??= keySlice.ToString();
-
-                        var inner = new InvalidOperationException(
-                            $"The expiration date format for compare exchange '{CompareExchangeKey.SplitStorageKey(storageKey).Key}' is not valid: '{expirationDate}'. Use the following format: {DateTime.UtcNow:O}");
-                        throw new RachisApplyException("Could not apply command.", inner);
-                    }
-
-                    var expiry = date.ToUniversalTime();
-                    ticks = expiry.Ticks;
-                    return true;
-                }
-                else
-                {
-                    storageKey ??= keySlice.ToString();
-                    var inner = new InvalidOperationException(
-                        $"The type of {Constants.Documents.Metadata.Expires} metadata for compare exchange '{CompareExchangeKey.SplitStorageKey(storageKey).Key}' is not valid. Use the following type: {nameof(DateTime)}");
-                    throw new RachisApplyException("Could not apply command.", inner);
-                }
-            }
-
-            return false;
+            var date = DateTime.ParseExact(expirationDate, DefaultFormat.DateTimeFormatsToRead, CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind);
```

Comment:
Wouldn't be safer to use `DateTime.TryParseExact()` as we did previously? I know that other value isn't supposed to be under @expires metadata but just in case ...

Reply:
I want it to throw here to make the difference between it didn't find and it didn't succeed to parse (and the previous implementation throw if it didn't succeed to parse).
For add/delete command I am catching the exception and rethrow with more information.
When deleting on expired, it shouldn't happen and in this case, there is a bug (there are guards to disallow modify the @expires with no date value).
I can catch it and rethrow with more information for debugging propose.

Reply:
If that was indented and there is good reason then it's okay.